src/createServer.js

```diff
@@ -1,3 +1,53 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase/convertToCase');
+const { parseURL } = require('./parseURL');
+const errorMessages = require('./errorMessages');
+
+const createServer = () => {
+  return http.createServer((req, res) => {
+    const { stringCase, queryString } = parseURL(req.url, req.headers.host);
+    const validation = {
+      errors: [],
+    };
+
+    res.writeHead(200, { 'Content-Type': 'application/json' });
+
+    if (!queryString) {
+      validation.errors.push({
+        message: errorMessages.TEXT_REQUIRED,
+      });
+    }
+
+    if (!stringCase) {
+      validation.errors.push({
+        message: errorMessages.TOCASE_REQUIRED,
+      });
+    }
+
+    let result;
+
+    try {
+      result = {
+        ...convertToCase(queryString, stringCase),
+        originalText: queryString,
+        targetCase: stringCase,
+      };
+    } catch {
+      if (stringCase) {
+        validation.errors.push({
+          message: errorMessages.INVALID_CASE,
+        });
+      }
+    }
+
+    if (!validation.errors.length) {
+      res.end(JSON.stringify(result));
+    } else {
+      res.statusCode = 400;
+      res.end(JSON.stringify(validation));
+    }
+  });
```

Comment:
Create a separate function to send response
```suggestion
      if (validation.errors.length) {
        sendResponse(400, result);
        
        return;
     }
     
      const result = {
        ...convertToCase(queryString, stringCase),
        originalText: queryString,
        targetCase: stringCase,
      }
      
     sendResponse(200, result);
  });
```